Keypad.c

```diff
@@ -0,0 +1,223 @@
+   P5 -> REN = (C1|C2|C3|C4);                   // use enable resistor
+   P5 -> OUT &= ~(C1|C2|C3|C4);                 //  pull down resistors
+
+
+   P2 -> DIR = (R0|R1|R2|R3);                   //sets rows as outputs
+/*
+   P2 -> SEL1 &= ~(R0|R1|R2|R3);                //Not sure why but it doesn't work as well with this section of code
+   P2 -> SEL0 &= ~(R0|R1|R2|R3);
+   P2 -> OUT |= (R0|R1|R2|R3);                  // sets it high
+*/
+   __enable_irq();
+   NVIC -> ISER[1] = 1 << ((PORT5_IRQn) & 31);
+}
+void PORT5_IRQHandler(void)                     //interrupt handler on Port 5
+{
+   P5->IFG &= ~(C1|C2|C3|C4);                   //clear interrupt flag
+   keypad_setkey();
+}
+
+void keypad_setkey(void)                        //function that checks the rows and columns for which button was pressed
+{
+    P2 -> OUT &= ~(R0|R1|R2|R3);                //set all low
+    uint8_t COL = 0;
+
+    P2 -> OUT = R0;
+    delay_ms(25, CLK);
+    COL = P5 -> IN & (C1|C2|C3|C4);
+    if (COL != 0)                               //Multiple if statements that goes through the rows to select the correct key based on column
+    {
+        if ((COL & C1) != 0)
+        {
+            key = 1;
+        }
+        else if ((COL & C2) !=0)
+        {
+            key = 2;
+        }
+        else if ((COL & C3) != 0)
+        {
+            key = 3;
+        }
+        else
+        {
+            key = 65;
+        }
+    }
+
+    P2 -> OUT = R1;
+    delay_ms(25, CLK);
+    COL = P5 -> IN &(C1|C2|C3|C4);
+    if (COL != 0)
+    {
+        if ((COL & C1) != 0)
+        {
+            key = 4;
+        }
+        else if ((COL & C2) != 0)
+        {
+            key = 5;
+        }
+        else if ((COL & C3) != 0)
+        {
+            key = 6;
+        }
+        else
+        {
+            key = 66;
+        }
+    }
+
+    P2 -> OUT = R2;
+    delay_ms(25, CLK);
+    COL = P5 -> IN &(C1|C2|C3|C4);
+    if (COL != 0)
+    {
+        if ((COL & C1) != 0)
+        {
+            key = 7;
+        }
+        else if ((COL & C2) != 0)
+        {
+            key = 8;
+        }
+        else if ((COL & C3) != 0)
+        {
+            key = 9;
+        }
+        else
+        {
+            key = 67;
+        }
+    }
+
+    P2 -> OUT = R3;
+    delay_ms(25, CLK);
+    COL = P5 -> IN &(C1|C2|C3|C4);
+    if (COL != 0)
+    {
+        if ((COL & C1) != 0)
+        {
+            key = 42;
+        }
+        else if ((COL & C2) != 0)
+        {
+            key = 0;
+        }
+        else if ((COL & C3) != 0)
+        {
+            key = 35;
+        }
+        else
+        {
+            key = 46;
+        }
+    }
+
+}
+uint8_t GetKey(void)                            //function that returns the key value
+{
+    uint8_t tempkey = key;
+    key = ResetKey;
+    return tempkey;
+}
+
+void keypad_testkey(void)                      //test function to write key value to terminal
+{
+ switch(GetKey())
+     {
+        case 1:
```

Comment:
When you have multiple statements in a case, it's best to put brackets around them.